crates/ruma-events/src/member_hints.rs

```diff
@@ -0,0 +1,110 @@
+//! Types for Matrix member hint state events ([MSC4171]).
+//!
+//! This implements `m.member_hints` state event described in [MSC4171].
+//!
+//! [MSC4171]: https://github.com/matrix-org/matrix-spec-proposals/pull/4171
+
+use std::collections::BTreeSet;
+
+use ruma_common::OwnedUserId;
+use ruma_macros::EventContent;
+use serde::{Deserialize, Serialize};
+
+use crate::EmptyStateKey;
+
+/// The content for an `m.member_hints` state event.
+///
+/// Any users (service members) listed in the content should not be considered when computing the
+/// room name or avatar based on the member list.
+#[derive(Clone, Debug, Default, Serialize, Deserialize, EventContent, PartialEq)]
+#[ruma_event(type = "io.element.functional_members", kind = State, state_key_type = EmptyStateKey)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+pub struct MemberHintsEventContent {
+    /// The list of user IDs that should be considered a service member of the room.
+    pub service_members: BTreeSet<OwnedUserId>,
```

Comment:
We should probably add a constructor. Maybe with this signature would be helpful:

```rust
pub fn new(service_members: impl IntoIterator<Item = OwnedUserId>) -> Self
```
Or just a `BTreeSet` for simplicity, either is fine.

Reply:
5e568a78dac7ad8df5e3a58a3727b4763d1dfc43